lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -11,27 +11,46 @@
     // allow unit tests to disable caching behavior for testing convenience but have this always `true` in real scenarios
     platform.useCaches = true;
 
-    if (platform.localeLookupCache === undefined) {
-        platform.localeLookupCache = new platform.Map();
-    }
-    if (platform.localeBestFitCache === undefined) {
-        platform.localeBestFitCache = new platform.Map();
-    }
-
     // determine what backing library we are using
     // making these vars in JS allows us to more change how we
     // determine the backing library
-    let isPlatformUsingICU = !platform.winglob;
-    let isPlatformUsingWinGlob = platform.winglob;
+    const isPlatformUsingICU = !platform.winglob;
+    const isPlatformUsingWinGlob = platform.winglob;
 
     // constants
-    const NOT_FOUND = "NOT_FOUND";
+    const LANGUAGE_NOT_FOUND = "NOT_FOUND";
 
     // Built-Ins
     var setPrototype = platform.builtInSetPrototype;
     var getArrayLength = platform.builtInGetArrayLength;
     var callInstanceFunc = platform.builtInCallInstanceFunction;
 
+    // Helper for our extensive usage of null-prototyped objects
+    const bare = (obj = {}) => setPrototype(obj, null);
+
+    class IntlCache {
+        constructor(n = 25) {
+            this.n = n;
+            this._cache = bare();
+        }
+
+        get(key) {
+            return platform.useCaches ? this._cache[key] : undefined;
```

Comment:
Can `platform.useCaches` be changed at runtime? If not, might make sense to cache the value in `this` if the overhead of fetching the value costs anything.

Reply:
It can (which is why we added it) -- see full\unittest\IntlCore\IntlPlatformTests.js. Is there a way we can give an argument to ch/jshost to enable or disable this, and then we can set the value in native code based on that and itll never change after?